include/clang/Basic/DiagnosticASTKinds.td

```diff
@@ -287,4 +287,6 @@ def err_odr_non_type_parameter_type_inconsistent : Warning<
   "non-type template parameter declared with incompatible types in different "
   "translation units (%0 vs. %1)">, InGroup<OdrDiags>;
 def err_unsupported_ast_node: Warning<"cannot import unsupported AST node %0">;
+def warn_ast_importer_missing_decl_in_decl_context : Warning<
+  "Missing Decl in imported DeclContext. Decl: %0">;
```

Comment:
Maybe it would be great to also emit a source location in case it is valid to aid debugging?

Reply:
Yeah good idea, first I dumped the whole AST node with Decl::dump(), but that is just too much.